### PR TITLE
[Feature] add transmissionRiskLevel

### PR DESCRIPTION
**Pull request by thomasaugsten**

### CODE REVIEW

**Review by ChristianKienle — APPROVED** [review]

**Review by inf2381 — APPROVED** [review]

**Comment by thomasaugsten** [comment]
@ChristopherSchmitz Can you please approve the implementation

**Comment by steins-code** [comment]
@thomasaugsten Can this PR be merged?

**Comment by thomasaugsten** [comment]
Please wait for the approval of @ChristopherSchmitz

**Review by ChristopherSchmitz — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Services/ExposureSubmissionService.swift [hunk 1]
```diff
@@ -179,11 +179,28 @@ class ENAExposureSubmissionService: ExposureSubmissionService {
 				return
 			}
 
-			guard let keys = keys, !keys.isEmpty else {
+			guard var keys = keys, !keys.isEmpty else {
```
Comment (ChristopherSchmitz) [inline src/xcode/ENA/ENA/Source/Services/ExposureSubmissionService.swift:182]: @thomasaugsten If keys are not ordered desc by creation time in the array the code would:
- cut the wrong keys
- assign the wrong transmissionRisk defaults.

To be on the safe side I recommend to order the diagnosisKeys by creationtimestam

**Review by ChristopherSchmitz — CHANGES_REQUESTED** [review]
please order the TEK desc by time

File: src/xcode/ENA/ENA/Source/Services/ExposureSubmissionService.swift (hunk 1 above)
Comment (thomasaugsten) [reply src/xcode/ENA/ENA/Source/Services/ExposureSubmissionService.swift:182]: there are sorted by _rollingStartNumber.  There is no timestamp. I would prefer to not reorder the keys

**Review by thomasaugsten — COMMENTED** [review]
Sorting should before trimming to 14 values

**Review by ChristopherSchmitz — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Services/ExposureSubmissionService.swift [hunk 2]
```diff
@@ -190,11 +190,32 @@ class ENAExposureSubmissionService: ExposureSubmissionService {
...
+				[5, 6, 7, 8, 7, 5, 3, 2, 1, 1, 1, 1, 1, 1, 1]
+			}
+
+			keys.sort {
+				$0.rollingStartNumber > $1.rollingStartNumber
+			}
+			
+			if keys.count > 14 {
+				keys = Array(keys[0 ..< 14])
+			}
```
Comment (ChristianKienle) [inline src/xcode/ENA/ENA/Source/Services/ExposureSubmissionService.swift:208]:
```suggestion
			keys = keys.prefix(14)
```

File: src/xcode/ENA/ENA/Source/Services/ExposureSubmissionService.swift [hunk 3]
```diff
@@ -190,11 +190,32 @@ class ENAExposureSubmissionService: ExposureSubmissionService {
...
+			
+			let startIndex = 0
+			for i in startIndex...keys.count - 1 {
+				if i + 1 <= transmissionRiskDefaultVector.count - 1 {
+					keys[i].transmissionRiskLevel = UInt8(transmissionRiskDefaultVector[i + 1])
+				} else {
+					keys[i].transmissionRiskLevel = UInt8(1)
+				}
+			}
```
Comment (ChristianKienle) [inline src/xcode/ENA/ENA/Source/Services/ExposureSubmissionService.swift:217]:
```suggestion
			
			for (i, key) in keys.enumerated() {
				if i + 1 <= transmissionRiskDefaultVector.count - 1 {
					key.transmissionRiskLevel = UInt8(transmissionRiskDefaultVector[i + 1])
				} else {
					key.transmissionRiskLevel = UInt8(1)
				}
			}
```

**Pull request merged**